packages/view/backend/sirius-components-view-emf/src/main/java/org/eclipse/sirius/components/view/emf/diagram/providers/ViewInitialDirectEditElementLabelProvider.java

```diff
@@ -172,7 +167,7 @@ private Optional<LabelEditTool> getLabelEditTool(org.eclipse.sirius.components.v
     }
 
     private Optional<NodeDescription> getNodeDescription(List<NodeDescription> nodeDescriptions, String descriptionId) {
-        var optionalNodeDescription = nodeDescriptions.stream().filter(nodeDescription -> descriptionId.equals(this.idProvider.apply(nodeDescription))).findFirst();
+        var optionalNodeDescription = nodeDescriptions.stream().filter(nodeDescription -> descriptionId.equals(this.idProvider.getId(nodeDescription))).findFirst();
```

Comment:
This fixes the issue only partially. If for some reason the descriptionId we look for is not found, we will still end up in an infinite loop.

We should either add a guard at the beginning:
```java
if (nodeDescriptions.isEmpty()) {
  return Optional.empty();
}
```
or avoid calling ourselves recursively with an empty `childrenDescription` on line 175.